app/models/item.js

```diff
@@ -0,0 +1,30 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+var Types = Schema.Types;
+var ObjectId = Types.ObjectId;
+
+var QueryPlugin = require('mongoose-query');
+
+var ItemSchema = new Schema({
+  barcode : { type: String, unique:true },
```

Comment:
i think this cannot be unique because there might be multiple items which doesnt have barcode (=empty/na) and then its not unique anymore...